src/features/reviews/components/MovieDetailsDrawer.vue

```diff
@@ -249,6 +279,47 @@ watch(
   },
 );
 
+// Check if description text exceeds 2 lines
+const checkDescriptionHeight = async () => {
+  await nextTick();
+
+  if (!visibleTextRef.value || !fullTextRef.value) {
+    shouldShowReadMore.value = false;
+    return;
+  }
+
+  // Calculate line height from the visible element
+  const computedStyle = window.getComputedStyle(visibleTextRef.value);
+  const lineHeight = parseFloat(computedStyle.lineHeight);
+
+  // Get the full height of the text
+  const fullHeight = fullTextRef.value.scrollHeight;
+
+  // Check if full height exceeds 2 lines (with small tolerance)
+  const twoLineHeight = lineHeight * 2;
+  shouldShowReadMore.value = fullHeight > twoLineHeight + 2; // 2px tolerance
+};
+
+// Watch for movie changes and reset description state
+watch(
+  () => props.movie?.id,
+  () => {
+    isDescriptionExpanded.value = false;
+    void checkDescriptionHeight();
+  },
+  { immediate: true },
+);
+
+// Also check when drawer opens
+watch(
+  () => props.isOpen,
```

Comment:
`isOpen` doesn't affect the layout. We don't need this watch if we key the component to change on a new movie. We should just check the height in an `onMounted` hook